package.json

```diff
@@ -1,10 +1,14 @@
 {
   "private": true,
   "devDependencies": {
+    "@chialab/esbuild-plugin-meta-url": "^0.17.5",
+    "@web/rollup-plugin-import-meta-assets": "^1.0.7",
```

Comment:
Can you explain in a little more detail why these plugins are needed?

What breaks if they are not included?   Is this something we should be looking to fix in emscripten?

Reply:
Unlike webpack, esbuild and rollup require a plugin to handle referencing assets with `new URL(..., import.meta.url)`.  These ensure that the `.wasm` file gets copied to the bundle directory and that the Emscripten runtime code is rewritten to point to it.  Of course, you can also do this manually as is done in the CommonJS webpack test, but it's probably good to take advantage of the asset bundling (and also know if it breaks for some reason).

As I mentioned in the issue, esbuild may add built-in support for `import.meta.url` assets, but the current implementation doesn't work properly for WebAssembly (whereas the plugin works, at least for simple cases like this).

There's nothing that needs to be fixed in emscripten for the moment, aside from adding a leading './` to the path to the `.wasm` file (which isn't strictly necessary but will help ensure compatibility)

Reply:
When you say "you can also do this manually as is done in the CommonJS webpack test", does that mean these new tests are actually testing more than the existing test and the the existing test doesn't correctly bundle the wasm file?  Should the existing test be changed in some way to bundle the wasm file?

Also, just to confirm these plugins relate the bundling of the wasm and therefore would not be needed in the case of `-sSINGLE_FILE`?

Reply:
The existing test is fine, it's just that webpack is unable to automatically bundle wasm files referenced from CommonJS modules unless you `require` them, which is not a good idea since it makes the resulting module webpack-specific.  Usually I use [copy-webpack-plugin](https://webpack.js.org/plugins/copy-webpack-plugin/) to copy it into place as part of the webpack build.

When you use ES6 modules, you can use `import.meta.url` which allows the code to reference the accompanying wasm file in the same way from the browser, node, and webpack.

Yes, none of this is needed for `-sSINGLE_FILE` - which reminds me that I had some other issues with `-sSINGLE_FILE` that I could submit a PR for!

Reply:
(example of using copy-webpack-plugin: https://github.com/dhdaines/webpack-wasm-test/blob/main/webpack.config.cjs.js)